internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_providers.go

```diff
@@ -216,214 +212,26 @@ func PluralDataSource() *schema.Resource {
 }
 func dataSourceMongoDBAtlasFederatedSettingsIdentityProvidersRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 
 	federationSettingsID, federationSettingsIDOk := d.GetOk("federation_settings_id")
 
-	options := &matlas.ListOptions{
-		PageNum:      d.Get("page_num").(int),
-		ItemsPerPage: d.Get("items_per_page").(int),
-	}
-
 	if !federationSettingsIDOk {
 		return diag.FromErr(errors.New("federation_settings_id must be configured"))
 	}
+	// There is no available paginated method
+	// If no protocol is specified, only SAML identity providers will be returned. Use ListIdentityProvidersWithParams to specify protocol
```

Comment:
q: when supporting OIDC would this call have to be adjusted? Would we require adding a new attribute to this data source or will the ListIdentityProviders return both protocols? Worth taking into account for the scope document in case changes are needed.

Reply:
My understanding here is that for read of plural data source, we will keep same attributes, and return both protocols(SAML & OIDC). So yes, call would need to be adjusted to call both protocols.